client-spec/client-specification-v1.md

```diff
@@ -0,0 +1,322 @@
+
+### name
+
+This field contains a user-friendly name that can be used to present the mod to the user.
+This field is not necessarily unique;
+
+clients must not identify a package using its `name`.
+However, it is recommended to use this field to help users
+search for the identifier of a module.
+
+### abstract
+
+This field contains a short description of what a module contains.
+Clients should use the value of this field to provide a description to the user, if any.
+
+### license
+
+This field is required by the metadata specification.
+Clients are not prescribed any behaviour regarding the value of this field.
+
+
+## Optional Fields
+
+The following fields are not required by the specification.
+Clients should not assume they are present in valid metadata;
+however, if they are present, clients must implement
+the behaviours prescribed in the following sections.
+
+### description
+
+The description field contains a longer description of the module
+than the one provided in the `abstract` field.
+Clients may use this field to display a verbose description of the module.
+
+### comment
+
+A `comment` field may appear at any point of the metadata.
+These fields will always contain a string.
+The value of a `comment` should not be displayed to the user,
+but may be employed for any purpose.
+
+### author
+
+Either a string, or a list of strings.
+The content of the `author` field should be shown to the user,
+and it should be used to search for specific modules if present.
+
+### download_size
+
+Size (in bytes) of the file downloaded through the `download` link.
+Showing the total download size to the user before any download
+is recommended, but not required.
+
+### release_status
+
+This field can have the following values:
+- `stable`
+- `testing`
+- `development`
+
+This field may be use to restrict the list of installable packages
+(e.g, to only allow `stable` modules).
+
+If this field is missing, clients must assume its value to be `stable`.
+
+### ksp_version
+
+This field may contain either:
+- `any`
+- a string matching the regular expression:
+		[0-9]+\\.[0-9]+(\\.[0-9]+)?
+
+If the field contains the string `any`,
+clients must allow the installation of the module
+on any version of the game,
+unless the `ksp_version_min` and `ksp_version_max` are also present:
+in this case, the client must respect these constraints.
+
+Otherwise, the client must only allow the installation of the mod
+into a copy of the game with a matching version number.
+
+### ksp_version_min and ksp_version_max
+
+Like the `ksp_version` field, these fields can contain either the string `any`
+or a version number.
+
+If `ksp_version` is `any` and either of these fields is defined,
+the client must enforce these constraints.
+
+If `ksp_version` contains a specific version number,
+these two fields should not be present.
+In this case, the client must disregard their value and only enforce
+the most specific version constraint.
+
+
+## install stanzas
```

Comment:
"If no install sections are provided, a CKAN client must find the top-most directory in the archive that matches the module identifier, and install that with a target of GameData." fetched from https://github.com/KSP-CKAN/CKAN/blob/master/Spec.md#install